Expand open container api on HumanEntity to support custom titles

Sytm opened the pull request:
The methods now take an optional component that overrides the title of the container.

I tested these changes with every method and every container type accepts the new overridden title just fine with the villager / merchant being the exception because the level of the villager is appended to the name

Machine-Maker commented on patches/api/0213-Add-additional-open-container-api-to-HumanEntity.patch:
```diff
@@ -3,16 +3,134 @@ From: JRoy <joshroy126@gmail.com>
+      */
+     @Nullable
+-    public InventoryView openMerchant(@NotNull Merchant merchant, boolean force);
++    public InventoryView openMerchant(@NotNull Merchant merchant, boolean force, @Nullable net.kyori.adventure.text.Component name);
```
Use `title` to maintain consistency.

Machine-Maker commented on patches/api/0213-Add-additional-open-container-api-to-HumanEntity.patch:
```diff
@@ -3,16 +3,134 @@ From: JRoy <joshroy126@gmail.com>
++     * opened.
++     */
++    @Nullable
++    public InventoryView openMerchant(@NotNull Villager trader, boolean force, @Nullable net.kyori.adventure.text.Component name);
```
Use `title` to maintain consistency.

Sytm replied:
My rationale for calling this parameter name instead of title was that when you set the title override to "Test" you would see "Test - Novice" instead of just "Test".

I think it is better to add some additional JavaDoc explaining that.

Should I still rename the param?

---
Yeah nvm, setting the villager level to 0 removes the suffix

Machine-Maker commented on patches/server/0464-Add-additional-open-container-api-to-HumanEntity.patch:
```diff
@@ -3,47 +3,148 @@ From: JRoy <joshroy126@gmail.com>
+ 
+     @Override
+-    public OptionalInt openMenu(@Nullable MenuProvider factory) {
++    public OptionalInt openMenu(@Nullable MenuProvider factory, @Nullable net.kyori.adventure.text.Component titleOverride) { // Paper - overridable title
```
I don't like calling this `titleOverride`. Even tho its just a param name, it should be something like `defaultTitleOverride` or something to distinguish it from the actual title override just added to InventoryOpenEvent

Machine-Maker commented on patches/server/0465-Add-additional-open-container-api-to-HumanEntity.patch:
```diff
@@ -3,47 +3,155 @@ From: JRoy <joshroy126@gmail.com>
+ 
+     @Override
+-    public OptionalInt openMenu(@Nullable MenuProvider factory) {
++    public OptionalInt openMenu(@Nullable MenuProvider factory, @Nullable net.kyori.adventure.text.Component defaultTitleOverride) { // Paper - overridable title
```
I think instead of this diff, you can just create new instances of SimpleMenuProvider and pass in the new Component title. That way it avoids all the diff in this methods and doesn't add yet another local var that is related to a title.

Machine-Maker commented on patches/server/0465-Add-additional-open-container-api-to-HumanEntity.patch:
```diff
@@ -3,47 +3,155 @@ From: JRoy <joshroy126@gmail.com>
++            name = io.papermc.paper.adventure.PaperAdventure.asVanilla(title);
++            level = 0; // Set level to 0 to remove title suffix
++        }
++        // Paper end
```
Not a fan of this because it just overwrites what they entity might have the level set to. I didn't know the title came with a suffix. Calling it `name` then is fine since its not the full title. People can manipulate the `Merchant` if they want to change the level. (I see there's no API for that for the custom merchants, just the entities but that can be added)

Sytm replied:
For the CustomMerchant that is irrelevant because they never append a suffix at all

Machine-Maker reviewed (changes requested):
We don't have to stick with upstream's ridiculousness in adding redundant `public` keywords to interface methods. For new methods, just don't include that.

Sytm committed:
Expand open container api on HumanEntity

The methods now take an optional adventure component that overrides the title of the container
